test/distributed/org/apache/cassandra/distributed/upgrade/MixedModeReadTest.java

```diff
@@ -51,7 +51,7 @@ public void mixedModeReadColumnSubsetDigestCheck() throws Throwable
             // we need to let gossip settle or the test will fail
             int attempts = 1;
             //noinspection Convert2MethodRef
-            while (!((IInvokableInstance) (cluster.get(1))).callOnInstance(() -> Gossiper.instance.isUpgradingFromVersionLowerThan(CassandraVersion.CASSANDRA_4_0.familyLowerBound.get()) &&
+            while (!((IInvokableInstance) (cluster.get(1))).callOnInstance(() -> Gossiper.instance.isUpgradingFromVersionLowerThan(CassandraVersion.CASSANDRA_4_0) &&
```

Comment:
Nit: we can remove a couple of parenthesis and the `Convert2MethodRef` inspection suppression, which I think doesn't apply here:
```suggestion
            while (!((IInvokableInstance) cluster.get(1)).callOnInstance(() -> Gossiper.instance.isUpgradingFromVersionLowerThan(CassandraVersion.CASSANDRA_4_0) &&
```